plugin/src/main/java/io/jenkins/plugins/remotingkafka/GlobalKafkaConfiguration.java

```diff
@@ -21,33 +47,216 @@ public static GlobalKafkaConfiguration get() {
         return GlobalConfiguration.all().get(GlobalKafkaConfiguration.class);
     }
 
-    public String getConnectionURL() {
-        return connectionURL;
+    public String getBrokerURL() {
+        return brokerURL;
     }
 
     public String getZookeeperURL() {
         return zookeeperURL;
     }
 
-    public FormValidation doCheckConnectionURL(@QueryParameter String connectionURL) {
-        if (StringUtils.isBlank(connectionURL)) {
-            return FormValidation.warning(Messages.GlobalKafkaConfiguration_KafkaConnectionURLWarning());
+    public String getUsername() {
+        return username;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public String getSslTruststoreLocation() {
+        return sslTruststoreLocation;
+    }
+
+    public String getSslTruststorePassword() {
+        return sslTruststorePassword;
+    }
+
+    public String getSslKeystoreLocation() {
+        return sslKeystoreLocation;
+    }
+
+    public String getSslKeystorePassword() {
+        return sslKeystorePassword;
+    }
+
+    public String getSslKeyPassword() {
+        return sslKeyPassword;
+    }
+
+    public String getCredentialsId() {
+        return credentialsId;
+    }
+
+    public FormValidation doCheckBrokerURL(@QueryParameter("brokerURL") final String brokerURL) {
+        if (StringUtils.isBlank(brokerURL)) {
+            return FormValidation.error(Messages.GlobalKafkaConfiguration_KafkaConnectionURLWarning());
         }
         return FormValidation.ok();
     }
 
-    public FormValidation doCheckZookeeperURL(@QueryParameter String zookeeperURL) {
+    public FormValidation doCheckZookeeperURL(@QueryParameter("zookeeperURL") String zookeeperURL) {
         if (StringUtils.isBlank(zookeeperURL)) {
-            return FormValidation.warning(Messages.GlobalKafkaConfiguration_ZookeeperURLWarning());
+            return FormValidation.error(Messages.GlobalKafkaConfiguration_ZookeeperURLWarning());
+        }
+        return FormValidation.ok();
+    }
+
+    public FormValidation doCheckUsername(@QueryParameter("username") String username) {
```

Comment:
`doCheck*` methods must always be annotated with `@RequirePOST`. Otherwise they may be used for CSRF attacks: https://www.owasp.org/index.php/Cross-Site_Request_Forgery_(CSRF)